docs/latest/design/request-authentication.md

```diff
@@ -0,0 +1,537 @@
+      virtual_hosts:
+        - name: default-eg-http-2
+          domains:
+            - '*'
+          routes:
+            - match:
+                prefix: /bar
+                headers:
+                  - name: ':authority'
+                    string_match:
+                      exact: www.example2.com
+              route:
+                cluster: default-backend2-rule-0-match-0-www.example2.com
+dynamic_active_clusters:
+  - cluster:
+      name: default-backend-rule-0-match-0-www.example.com
+      ...
+        endpoints:
+        - locality: {}
+          lb_endpoints:
+            - endpoint:
+                address:
+                  socket_address:
+                    address: $BACKEND_SERVICE1_IP
+                    port_value: 3000
+  - cluster:
+      '@type': type.googleapis.com/envoy.config.cluster.v3.Cluster
+      name: default-backend-rule-1-match-0-www.example.com
+      ...
+        endpoints:
+        - locality: {}
+          lb_endpoints:
+            - endpoint:
+                address:
+                  socket_address:
+                    address: $BACKEND_SERVICE2_IP
+                    port_value: 3000
+...
+```
+
+__Note:__ The JWT provider cluster and route is omitted from the above example for brevity.
+
+### Implementation Outline
+
+* Create a `security` API group and add the Authentication API type to this group.
+* Update the Kubernetes provider to get/watch Authentication resources that are referenced by managed HTTPRoutes. Add
+  the referenced Authentication object to the resource map and publish it.
+* Update the resource translator to include the Authentication API in HTTPRoute processing.
+* Update the xDS translator to translate an Authentication into xDS resources. The translator should perform the
+  following:
+  * Convert a list of JWT rules from the xds IR into an Envoy JWT filter config.
+  * Create a JWT authentication filter.
+  * Build the HTTP Connection Manager (HCM) HTTP filters.
+  * Build the HCM.
+  * When building the Listener, create an HCM for each filter-chain.
+
+## Adding Authentication Provider Types
+
+Additional authentication provider types can be added in the future through the `ProviderType` API. For example, to add
+the `Foo` authentication provider type:
+
+Define the `FooProvider` type:
+
+```go
+package v1alpha1
+
+// FooProvider defines the "Foo" authentication provider type.
+type FooProvider struct {
+	// TODO: Define fields of the Foo authentication provider type.
+}
+```
+
+Add the `FooProvider` type to `AuthenticationSpec`:
+
+```go
+package v1alpha1
+
+type AuthenticationSpec struct {
+	...
+	
+	// Foo defines the Foo authentication type. For additional
+	// details, see:
+	//
+	//   <INSERT_LINK>
+	//
+	// +optional
+	Foo *FooAuthentication `json:"foo"`
+}
+```
+
+Authentication should support additional authentication types in the future, for example:
+- mutualTLS (client certificate)
+- OAuth2
+- OIDC
+- External authentication
```

Comment:
prefer if we removed `external authentication` for now, this falls under `expert-mode`, and could be achieved by configuring xds resources directly

Reply:
Done in commit [8c2fde2](https://github.com/envoyproxy/gateway/pull/529/commits/8c2fde2f616a38b96cf602e0d40d5c6ebc2bbe3e).